packages/maskbook/src/plugins/Ethereum/UI/GasNowDialog.tsx

```diff
@@ -0,0 +1,232 @@
+import { useMemo, useState, useCallback } from 'react'
+import BigNumber from 'bignumber.js'
+
+const useStyles = makeStyles((theme: Theme) => ({
+    content: {},
+    list: {
+        display: 'flex',
+        flexDirection: 'row',
+        justifyContent: 'space-between',
+        padding: theme.spacing(1),
+    },
+    box: {
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'space-between',
+        alignItems: 'baseline',
+        width: 166,
+        height: 160,
+        border: `2px solid ${theme.palette.mode === 'dark' ? '#2f3336' : '#EBEEF0'}`,
+        borderRadius: 12,
+        cursor: 'pointer',
+    },
+    disableBox: {
+        opacity: 0.4,
+        cursor: 'default',
+    },
+    boxSelected: {
+        border: '2px solid #1C68F3',
+    },
+    intro: {
+        color: '#7B8192',
+        padding: theme.spacing(1),
+        fontSize: 14,
+    },
+    title: {
+        fontSize: 18,
+        fontWeight: 300,
+    },
+    estimate: {
+        color: '#7B8192',
+        fontSize: 12,
+    },
+    gwei: {
+        fontSize: 18,
+        fontWeight: 400,
+        display: 'inline-block',
+    },
+    gweiBox: {
+        display: 'flex',
+        alignItems: 'center',
+        marginBottom: theme.spacing(0.75),
+    },
+    head: {
+        width: '100%',
+        display: 'flex',
+        justifyContent: 'space-between',
+        alignItems: 'center',
+    },
+    dot: {
+        width: 12,
+        height: 12,
+        color: theme.palette.mode === 'dark' ? '#2F3336' : '#EBEEF0',
+    },
+    dotSelector: {
+        color: '#1C68F3',
+    },
+    customInput: {
+        width: 60,
+        marginRight: theme.spacing(1),
+    },
+    button: {
+        margin: '16px auto',
+        minWidth: 260,
+    },
+}))
+
+export function GasNowDialog() {
+    const classes = useStyles()
+    const { t } = useI18N()
+    const gasNow = useValueRef(currentGasNowSettings)
+    const rapidGasFirstCache = useMemo(() => gasNow?.rapid, [])
+    const [customGas, setCustomGas] = useState(formatWeiToGwei(rapidGasFirstCache ?? 0).toString())
+    const customGasToWei = Number(customGas === '' ? 0 : customGas) * 10 ** 9
+    const customEstimateTime = gasNow
+        ? customGasToWei >= gasNow.rapid
+            ? t('plugin_gas_now_dialog_sec', { time: 15 })
+            : customGasToWei >= gasNow.fast
+            ? t('plugin_gas_now_dialog_min', { time: 1 })
+            : customGasToWei >= gasNow.standard
+            ? t('plugin_gas_now_dialog_min', { time: 3 })
+            : t('plugin_gas_now_dialog_min', { time: '>10' })
+        : ''
+    const { open, closeDialog, setDialog } = useRemoteControlledDialog(
+        EthereumMessages.events.gasPriceDialogUpdated,
+        (_ev) => void 0,
+    )
+    const { value: detailedTokens } = useAssets([])
+    const [select, setSelect] = useState(gasNow ? 1 : 2)
+    const nativeToken = detailedTokens[0]
```

Comment:
`useNativeTokenDetailed`

Reply:
`useNativeTokenDetailed` can not get the usd price.